### DISCUSSION TITLE
Disable on Block (Station Mode) is bypassed when running in Random Mode)

### DISCUSSION THREAD

**Discussion by TekyTeky** [Novakasa/brickrail#166] [General]
When I put a train into a Station - Disabled Block, it becomes active when I switch to Random.
Is this by design? If yes then that's ok.. I just wasn't expecting it.
Would be better if a train is disabled that it stays disabled until it is moved out of the block?

**Comment by Novakasa** [comment]
Oh yeah, this is not intended. I converted to an issue #167 
I didn't really think a lot about these cases, since I mainly thought about using them as separate, non-connected blocks like this:
![image](https://github.com/Novakasa/brickrail/assets/59559612/8464afa2-f552-47e0-94e2-335ce8495893)
I wanted to be able to disable trains, but didn't really wanted to create a new GUI area where they are stored, so I just thought of a layout area where you can drag them which is separate from the rest of the layout, so they won't be able to move in random mode anyway. The main advantage to this is that the train will not need to be turned on when you turn on control devices. This basically represents the train being "off the layout".

I guess when a block has the train disabled checkbox set while it is part of the layout, it becomes a bit weird when a non-disabled train arrives in this block, for example via random targets. Would you expect the train to be disabled then?

One thing I considered was to only be able to set the disabled checkbox when the block is not connected to anything else. How would you ideally use the disabled blocks?

**Reply by TekyTeky** [reply]
Ah.. this makes sense.
I had a block at the end of a line set to disable, so when a train was sent to that block they shut down. 
So I thought it would not be part of the random choice.
If a train is disabled will it eventually disconnect?
I was quite surprised when it started up again.

What you suggest is good, as long as the train disconnects, otherwise if I take it physically off the layout do I need to manually disconnect it before I turn it off?

I will have a play tomorrow.

**Reply by Novakasa** [reply]
I think I will make it so you can only check the disabled box when the block is separate from the layout, and yeah I should add that the hub will disconnect once the train is moved to a disabled block. Right now it doesn't do that.

**Reply by Novakasa** [reply]
This is now fixed, but I did not prevent disabling a block if it is connected.
If a train is in a disabled block the hub will disconnect. A disabled block will not be traversed in random mode, and any train in a disabled block will not choose a random target.
